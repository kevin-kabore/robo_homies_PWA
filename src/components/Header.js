import React, { Component } from 'react';
import CounterButton from './CounterButton';

const Header = () => {
	return (
		<div>
			<h1 className="f1">Robo Homies</h1>
			<CounterButton color={'red'} />
		</div>
	)
}

export default Header;