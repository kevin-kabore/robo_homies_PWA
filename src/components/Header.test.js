import { shallow } from 'enzyme';
import React from 'react';

import Header from './Header';

describe('Header Component', () => {
	it('Will render the Component', () => {
		// snapshot testing used for all presentational/stateless components the whole component
		expect(shallow(<Header />)).toMatchSnapshot()		
	})
})
