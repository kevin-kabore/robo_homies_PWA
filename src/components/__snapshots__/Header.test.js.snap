// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Will render the Component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="f1"
>
          Robo Homies
</h1>,
        <CounterButton
          color="red"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Robo Homies",
          "className": "f1",
        },
        "ref": null,
        "rendered": "Robo Homies",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color": "red",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="f1"
>
            Robo Homies
</h1>,
          <CounterButton
            color="red"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Robo Homies",
            "className": "f1",
          },
          "ref": null,
          "rendered": "Robo Homies",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "color": "red",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
